### PR TITLE
Uploader fixes

### PR DESCRIPTION
**Pull request by almirsarajcic**

### CODE REVIEW

File: lib/chat_web/live/main_live/layout/uploader.ex [hunk 1]
```diff
@@ -154,15 +210,17 @@ defmodule ChatWeb.MainLive.Layout.Uploader do
   defp entry(assigns) do
     ~H"""
     <div
-      class="flex mb-5 bg-white border-purple relative w-full z-0"
+      class={"flex mb-5 border-purple relative w-full z-0 " <> if(not @mobile? or @metadata.destination.type == @type and @metadata.destination.pub_key == @pub_key, do: "bg-white", else: "bg-pink-100")}
```
[the remark this answers is not in the document]
Comment (almirsarajcic) [reply lib/chat_web/live/main_live/layout/uploader.ex:213]: In the sidebar uploader, I didn't differentiate between current dialog/room uploads and the others.
I can change this.

File: lib/chat_web/live/main_live/layout/uploader.ex [hunk 2]
```diff
@@ -90,46 +106,84 @@ defmodule ChatWeb.MainLive.Layout.Uploader do
...
+              else: show_modal("restrict-write-actions")
+          }
+        >
+          <.icon
+            id="attach"
+            class={
+              classes("w-7 h-7 flex", %{"fill-red-500" => !@enabled, "fill-gray-400" => @enabled})
+            }
+          />
+          <.dropdown class="mr-5" id="attachFileDropdown">
```
[the remark this answers is not in the document]
Comment (almirsarajcic) [reply lib/chat_web/live/main_live/layout/uploader.ex:132]: Sorry, I forgot you mentioned we should have a different button layout than in that old commit.

Yes, in Chrome on Android, it shows images directly.

File: lib/chat_web/live/main_live/index.html.heex [hunk 3]
```diff
@@ -433,7 +433,12 @@
               </ul>
             </div>
 
-            <Layout.Uploader.uploader config={@uploads.file} uploads={@uploads_metadata} />
+            <Layout.Uploader.uploader
+              config={@uploads.file}
+              pub_key={@peer && @peer.pub_key}
```
Comment (sergey-lukianov) [inline lib/chat_web/live/main_live/index.html.heex:438]: `assigns[:peer][:pub_key]` does the same. Just less hack-ish way

File: lib/chat_web/live/main_live/index.html.heex [hunk 4]
```diff
@@ -684,7 +691,12 @@
               </div>
             </div>
 
-            <Layout.Uploader.uploader config={@uploads.file} uploads={@uploads_metadata} />
+            <Layout.Uploader.uploader
+              config={@uploads.file}
+              pub_key={@room && @room.pub_key}
```
Comment (sergey-lukianov) [inline lib/chat_web/live/main_live/index.html.heex:696]: `assigns[:room][:pub_key]`

File: lib/chat_web/live/main_live/layout/uploader.ex [hunk 5]
```diff
@@ -223,14 +272,31 @@ defmodule ChatWeb.MainLive.Layout.Uploader do
...
+          >
+            <.icon id="image" class="w-4 h-4 fill-grayscale" />
+            <span class="ml-2">Image</span>
+          </a>
+        </div>
+
         <input
-          accept="audio/*,image/*,video/*"
-          class="block p-2 flex flex-col items-center text-sm text-black/50 file:mr-4 file:py-2 file:px-4 file:rounded-full file:border-0 file:text-sm file:font-semibold file:bg-purple file:text-purple50 file:cursor-pointer"
+          accept="image/*"
```
Comment (sergey-lukianov) [inline lib/chat_web/live/main_live/layout/uploader.ex:298]: can we keep `video/*` here?  So it will show Android gallery w/ images and videos

Comment (almirsarajcic) [reply lib/chat_web/live/main_live/layout/uploader.ex:298]: Should I also add `audio/*` and change the button text from "Image" to "Media"?
https://github.com/Buckitup-chat/chat/pull/157#pullrequestreview-1285443640

Comment (buckitupnet) [reply lib/chat_web/live/main_live/layout/uploader.ex:298]: I support audio and "media" title

Comment (sergey-lukianov) [reply lib/chat_web/live/main_live/layout/uploader.ex:298]: If it opens gallery on Android then add
@Rus9688 said that `audio/*` breaks gallery behaviuor on Android

File: lib/chat_web/live/main_live/index.html.heex (hunk 3 above)
Comment (almirsarajcic) [reply lib/chat_web/live/main_live/index.html.heex:438]: That was my first approach, but it requires `Card` to implement Access protocol.
```
** (UndefinedFunctionError) function Chat.Card.fetch/2 is undefined (Chat.Card does not implement the Access behaviour. If you are using get_in/put_in/update_in, you can specify the field to be accessed using Access.key!/1)
    (chat 0.1.0) Chat.Card.fetch(#Chat.Card<name: "Mac", hash: "f04b0226f4bfb79363d0efc5a65671a08c34ea1eae5ec8374caf8e9c6b74f4e3", ...>, :pub_key)
```

File: lib/chat_web/live/main_live/layout/uploader.ex (hunk 5 above)
Comment (almirsarajcic) [reply lib/chat_web/live/main_live/layout/uploader.ex:298]: That's right. I've just tested it and even putting `image/*,video/*` didn't work, nor `.jpg,.jpeg,.png,.mp4`. The only way it opened gallery directly for me was `image/*`, so I won't change this in this PR. Let's experiment with that in #157.

File: lib/chat_web/live/main_live/index.html.heex (hunk 3 above)
Comment (sergey-lukianov) [reply lib/chat_web/live/main_live/index.html.heex:438]: https://hex.pm/packages/struct_access
and `use StructAccess` in Card

**Review by sergey-lukianov — APPROVED** [review]

**Pull request merged**